src/client.ts

```diff
@@ -1688,10 +1689,28 @@ export class StreamChat<StreamChatGenerics extends ExtendableGenerics = DefaultG
     );
   }
 
+  /**
+   * getUnreadCount - Returns unread counts for a single user
+   *
+   * @param {string} [userID] User ID.
+   *
+   * @return {<GetUnreadCountAPIResponse>}
+   */
   async getUnreadCount(userID?: string) {
     return await this.get<GetUnreadCountAPIResponse>(this.baseURL + '/unread', userID ? { user_id: userID } : {});
   }
 
+  /**
+   * getUnreadCountBatch - Returns unread counts for multiple users at once. Only works server side.
+   *
+   * @param {string[]} [userIDs] List of user IDs to fetch unread counts for.
+   *
+   * @return {<GetUnreadCountBatchAPIResponse>}
+   */
+  async getUnreadCountBatch(userIDs: string[]) {
+    return await this.post<GetUnreadCountBatchAPIResponse>(this.baseURL + '/unread_batch', { user_ids: userIDs });
```

Comment:
@JimmyPettersson85 just out of curiosity, why for a GET request is used POST method?

Reply:
For future proofing the backend we chose `POST` for this endpoint as we might hit querystring limits if we increase the limit of number of users. Currently the limit is 100 user IDs, and with a max length of 255 for a user ID that means a querystring of ~25k chars.

In going with `POST` we can handle > 100 user IDs in the future as well as support compression.